crates/bevy_reflect/src/reflect.rs

```diff
@@ -11,6 +11,22 @@ use std::{
 use crate::utility::NonGenericTypeInfoCell;
 pub use bevy_utils::AHasher as ReflectHasher;
 
+/// A simple enumeration of "kinds" of type, without any associated object.
+///
+/// A `ReflectKind` is obtained via [`Reflect::reflect_kind`].
```

Comment:
Maybe give a bit more context as to what a "kind" is? If we can give a sentence or two of detail or at least an example of what constitutes a "kind" that could be really useful.

And then I think `ReflectRef`, `ReflectMut`, and `ReflectOwned` should all link back to these docs when they mention "kind" in their own documentation.